compiler/hash-ast-desugaring/src/visitor.rs

```diff
@@ -463,51 +463,51 @@ impl<'s> AstVisitorMut for AstDesugaring<'s> {
 
     type TyFnRet = ();
 
-    fn visit_ty_fn(
+    fn visit_ty_fn_ty(
         &mut self,
         _: &Self::Ctx,
         _: hash_ast::ast::AstNodeRefMut<hash_ast::ast::TyFn>,
     ) -> Result<Self::TyFnRet, Self::Error> {
         Ok(())
     }
 
-    type TypeFnCallRet = ();
+    type TyFnCallRet = ();
 
     fn visit_ty_fn_call(
         &mut self,
         _: &Self::Ctx,
         _: hash_ast::ast::AstNodeRefMut<hash_ast::ast::TyFnCall>,
-    ) -> Result<Self::TypeFnCallRet, Self::Error> {
+    ) -> Result<Self::TyFnCallRet, Self::Error> {
         Ok(())
     }
 
-    type NamedTypeRet = ();
+    type NamedTyRet = ();
 
     fn visit_named_ty(
         &mut self,
         _: &Self::Ctx,
         _: hash_ast::ast::AstNodeRefMut<hash_ast::ast::NamedTy>,
-    ) -> Result<Self::NamedTypeRet, Self::Error> {
+    ) -> Result<Self::NamedTyRet, Self::Error> {
         Ok(())
     }
 
-    type RefTypeRet = ();
+    type RefTyRet = ();
 
     fn visit_ref_ty(
         &mut self,
         _: &Self::Ctx,
         _: hash_ast::ast::AstNodeRefMut<hash_ast::ast::RefTy>,
-    ) -> Result<Self::RefTypeRet, Self::Error> {
+    ) -> Result<Self::RefTyRet, Self::Error> {
         Ok(())
     }
 
-    type MergedTypeRet = ();
+    type MergedTyRet = ();
```

Comment:
An unrelated note, there should also be merging expressions in the ast, not just merging types..